tests/console/yast2_dns_server.pm

```diff
@@ -84,13 +85,13 @@ sub run {
     script_run 'yast2 dns-server', 0;
     assert_screen 'yast2-service-running-enabled';
     # Stop the service
-    send_key 'alt-s';
+    wait_screen_change { send_key 'alt-s' };
     assert_screen 'yast2-service-stopped-enabled';
     # Cancel yast2 to check the effect
-    wait_screen_change { send_key 'alt-c' };
-    if (check_screen('yast2-dns-server-quit')) {
-        wait_screen_change { send_key 'alt-y' };
-    }
+    # workaround for single send_key 'alt-c' because it doesn't work.
+    send_key_until_needlematch([qw(root-console yast2-dns-server-quit)], 'alt-c');
+    send_key 'alt-y';
+    assert_screen 'root-console';
```

Comment:
For other yast modules we have seen that sometimes it takes longer than 30 seconds for program to return. Here we also don't check exit code. Maybe we could combine these two things, wait for return code for 180 seconds as in other modules and then assert screens.